istioctl/cmd/workload.go

```diff
@@ -0,0 +1,402 @@
+	createCmd.PersistentFlags().StringSliceVarP(&labels, "labels", "l", nil, "The labels to apply to the workload instances; e.g. -l env=prod,vers=2")
+	createCmd.PersistentFlags().StringSliceVarP(&ports, "ports", "p", nil, "The incoming ports exposed by the workload instance")
+	createCmd.PersistentFlags().StringVarP(&serviceAccount, "serviceAccount", "s", "default", "The service identity to associate with the workload instances")
+	return createCmd
+}
+
+func generateWorkloadGroupYAML(u *unstructured.Unstructured, spec *networkingv1alpha3.WorkloadGroup) ([]byte, error) {
+	iSpec, err := unstructureIstioType(spec)
+	if err != nil {
+		return nil, err
+	}
+	u.Object["spec"] = iSpec
+
+	wgYAML, err := yaml.Marshal(u.Object)
+	if err != nil {
+		return nil, err
+	}
+	return wgYAML, nil
+}
+
+func configureCommand() *cobra.Command {
+	configureCmd := &cobra.Command{
+		Use:   "configure",
+		Short: "Generates all the required configuration files for a workload instance running on a VM or non-k8s environment",
+		Long: `Generates all the required configuration files for workload instance on a VM or non-k8s environment from a WorkloadGroup artifact.
+This includes a MeshConfig resource, the cluster.env file, and necessary certificates and security tokens.
+Configure requires either the WorkloadGroup artifact path or its location on the API server.`,
+		Example: `# configure example using a local WorkloadGroup artifact
+configure -f workloadgroup.yaml -o config
+
+# configure example using the API server
+configure --name foo --namespace bar -o config`,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if filename == "" && (name == "" || namespace == "") {
+				return fmt.Errorf("expecting a WorkloadGroup artifact file or the name and namespace of an existing WorkloadGroup")
+			}
+			if outputDir == "" {
+				return fmt.Errorf("expecting an output directory")
+			}
+			if clusterID == "" {
+				return fmt.Errorf("expecting a cluster id")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			wg := &clientv1alpha3.WorkloadGroup{}
+			kubeClient, err := kube.NewExtendedClient(kube.BuildClientCmd(kubeconfig, configContext), "")
+			if err != nil {
+				return err
+			}
+			if filename != "" {
+				if err := readWorkloadGroup(filename, wg); err != nil {
+					return err
+				}
+			} else {
+				wg, err = kubeClient.Istio().NetworkingV1alpha3().WorkloadGroups(namespace).Get(context.Background(), name, metav1.GetOptions{})
+				// errors if the requested workload group does not exist in the given namespace
+				if err != nil {
+					return fmt.Errorf("workloadgroup %s not found in namespace %s: %v", name, namespace, err)
+				}
+			}
+
+			if err = createConfig(kubeClient, wg, clusterID, revision, outputDir); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	configureCmd.PersistentFlags().StringVarP(&filename, "file", "f", "", "filename of the WorkloadGroup artifact")
```

Comment:
nit: can we add a comment about how if left empty, the WG will be pulled from the configured API server.